From 3734f4ae4816c372427f99ab5f0dfd7f386c8222 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 9 May 2012 11:05:12 +0200
Subject: [PATCH] plymouth/plymouth-pretrigger.sh: get consoledev from
 /sys/class/tty/console/active

---
 modules.d/50plymouth/plymouth-pretrigger.sh |    2 +-
 1 Datei geändert, 1 Zeile hinzugefügt(+), 1 Zeile entfernt(-)

diff --git a/modules.d/50plymouth/plymouth-pretrigger.sh b/modules.d/50plymouth/plymouth-pretrigger.sh
index 8fa0aec..483cc70 100755
--- a/modules.d/50plymouth/plymouth-pretrigger.sh
+++ b/modules.d/50plymouth/plymouth-pretrigger.sh
@@ -19,7 +19,7 @@ if [ -x /bin/plymouthd ]; then
 
         info "Starting plymouth daemon"
         mkdir -m 0755 /run/plymouth
-        consoledev=$(getarg console= | sed -e 's/,.*//')
+        read consoledev rest < /sys/class/tty/console/active
         consoledev=${consoledev:-tty0}
         [ -x /lib/udev/console_init -a -e "/dev/$consoledev" ] && /lib/udev/console_init "/dev/$consoledev"
         [ -x /bin/plymouthd ] && /bin/plymouthd --attach-to-session --pid-file /run/plymouth/pid
